### PR TITLE
Rfc doc

### PR DESCRIPTION
**Pull request by kudanai**
Opening PR for comments

### CODE REVIEW

**Comment by kudanai** [comment]
DO NOT MERGE: only here for the comments

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
+# Lottie Format Specification Contribution Guide
+
+This repository contains the specification text as well as Pull Requests
+with suggested improvements and contributions.
+
+The outline of this document closely follows contribution guidelines
+set out in the [GraphQL](https://github.com/graphql/graphql-spec) specification's process
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I don't think we need this

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+Contributions that do not change the interpretation of the spec but instead
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > spec

Should we use informal abbreviations? (Just a question not a request to change per se)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+to better understand expected and unacceptable behavior.
+
+
+## Contributing to Lottie Libraries
+
+A common point of confusion for those who wish to contribute to the Lottie Format is where
+to start. In fact, you may have found yourself here after attempting to make an
+improvement to a Lottie library. Should a new addition be made to the Lottie
+spec first or a Lottie library first? Admittedly, this can become a bit of a
+[chicken-or-egg](https://en.wikipedia.org/wiki/Chicken_or_the_egg) dilemma.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I don't think the link is needed

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+through a series of _stages_ intended to improve _visibility_, allow for
+_discussion_ to reach the best solution, and arrive at _consensus_. This process
+becomes ever more important as Lottie Animation Community Broadens.
+
+When proposing or weighing-in on any issue or pull request, consider the
+[Code of Conduct](https://github.com/lottie/lottie-spec/blob/main/8._Code_of_Conduct.md)
+to better understand expected and unacceptable behavior.
+
+
+## Contributing to Lottie Libraries
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: Probably this section should be later in the document?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+To allow a library to remain spec compliant while also implementing _proposals_
+and _drafts_, the library's maintainers may request that these new features are
+disabled by default with opt-in option flags or they may simply wait to merge a
+well-tested pull request until the spec proposal is _accepted_.
+
+## Workflow
+
+1. The primary workflow is centered around _this_ repository,
+   utilizing the following features.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: `:` instead of `.`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    * Pull Requests
+    * Issues
+    * Discussions
+
+2. Regular, documented meetings of the LAC working group, to review and progress proposals 
+   through the various stages as outlined below.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: Phrasing seems a bit off

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+
+Members of the community, who are interested in participating in discussions and other work 
+around the Lottie Format, be it the format itself, documentation, libraries, tools or 
+other services built around the format can do so around the repository.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: Seems an overly verbose way of saying "Contributions go here", perhaps we don't need this paragraph?

Comment (mbasaglia) [reply CONTRIBUTING.md]: Also let's stick with "Lottie format" without capitalization

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+In doing so, members can put forth _proposals_ for the format, which are 
+well reasoned formal recommendations or suggestions put toward consideration by 
+the community and working-group, aimed at fixing, improving or enhancing the
+Lottie Animation Format. Proposals can be initiated either via issues or pull requests
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > Lottie Animation Format

"Lottie format"

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+2. Regular, documented meetings of the LAC working group, to review and progress proposals 
+   through the various stages as outlined below.
+
+Members of the community, who are interested in participating in discussions and other work 
+around the Lottie Format, be it the format itself, documentation, libraries, tools or 
+other services built around the format can do so around the repository.
+
+In doing so, members can put forth _proposals_ for the format, which are 
+well reasoned formal recommendations or suggestions put toward consideration by 
+the community and working-group, aimed at fixing, improving or enhancing the
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: "working group" shouldn't he hyphened. Also aren't "improving" and "enhancing" synonyms?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+Lottie Animation Format. Proposals can be initiated either via issues or pull requests 
+(as detailed below). Any issue raised is considered an RFC, subject to review, 
+comments and deliberation by both community and working-group. An RFC may begin as an 
+issue or PR, and may not initially be reasoned out enough to be considered a formal 
+proposal (see below). However, it will eventually have to be detailed enough to be 
+considered a proposal.
+
+## Key stakeholders
+
+**Lottie Animation Working Group**
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: How is the working group different from LAC?

Also I'd use level 3 heading (`###`)  rather than bold text for these

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+
+A working group, put together by the LAC, is responsible for guiding the process, 
+moderating discussions, and so forth. The working group will conduct regular meetings to 
+review proposals, and either advance them through the process or put
+forward recommendations/suggestions for the same.
+
+Relevant details of the working-group, its members and tenure shall be published on
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: "working group"

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+
+**Lottie Animation Working Group**
+
+A working group, put together by the LAC, is responsible for guiding the process, 
+moderating discussions, and so forth. The working group will conduct regular meetings to 
+review proposals, and either advance them through the process or put
+forward recommendations/suggestions for the same.
+
+Relevant details of the working-group, its members and tenure shall be published on 
+the main repository.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: Was referred to as "this repository" above. Perhaps it should be consistent throughout. I like "main repository" better as it allows this document to be shared outside of the repo (eg: the docs site), it could be useful to have a link when the repository is mentioned the first time.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+
+**Lottie Sponsors**
+
+Individuals or organizations who are recognised by LAC are responsible for taking on proposals and guiding them through
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > guiding them

Sponsors don't guide the proposal but the submitter of the proposal, right?

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,233 @@
...
+
+A working group, put together by the LAC, is responsible for guiding the process, 
+moderating discussions, and so forth. The working group will conduct regular meetings to 
+review proposals, and either advance them through the process or put
+forward recommendations/suggestions for the same.
+
+Relevant details of the working-group, its members and tenure shall be published on 
+the main repository.
+
+**Lottie Sponsors**
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I'd remove "Lottie". Also I'm not sure if _Sponsors_ are a stakeholder group, this should be defined as part of the submission process. I think all sponsors are also editors

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+Individuals or organizations who are recognised by LAC are responsible for taking on proposals and guiding them through
+the process. Every proposal must have a sponsor attached to it, in order to progress through the process. A proposal may
+have multiple sponsors.
+
+Sponsor organizations shall be transparently documented on the main repository.
+
+**Editors**
+
+Either working-group members or selected contributors responsible to repository maintenance: i.e have write/edit access
+to the main repository.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: "working group". Also, the "i.e." part is redundant/

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+
+
+## Classifying Contributions
+
+Contributions can come in the form of issues or pull requests. Contributions through github discussions are not
+recognized. Discussions may be used for general discussion and brainstorming before it is put forth as a proposal and
+subsequently into the RFC process.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > Contributions through github discussions are not recognized

Phrasing is a bit misleading: they are not part of the standard but they can be part of the process. Perhaps "Contributions" should be changed to "Proposals" here?

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+
+Of contributions, they may be classified as follows:
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: Odd phrasing

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+Contributions that do not change the interpretation of the spec but instead
+improve legibility, fix editorial errors, clear up ambiguity and improve
+examples are encouraged and are often merged by a spec editor with little
+process.
+
+However, contributions that _do_ meaningfully change the interpretation of the
+spec must follow an RFC (Request For Comments) process led by a _sponsor_
+through a series of _stages_ intended to improve _visibility_, allow for
+_discussion_ to reach the best solution, and arrive at _consensus_. This process
+becomes ever more important as Lottie Animation Community Broadens.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: This stuff is repeated later, do we need to have it twice?

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,233 @@
...
+
+Of contributions, they may be classified as follows:
+
+1. **Contributions that _do not_ change the spec, or interpretation of the spec**  
+   These are issues and PRs largely aimed at improving legibility, fixing editorial errors, 
+   clearing up ambiguities, or adding examples to existing specifications. 
+   These may be progressed and merged without much process.
+
+2. **Contributions that _do_ meaningfully change the interpretation of the spec**  
+   These must progress through the stages of RFC, guided by a sponsor until
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > sponsor

I'd capitalize this "Sponsor" and add the definition somewhere around here rather than under stakeholders

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+1. **Contributions that _do not_ change the spec, or interpretation of the spec**  
+   These are issues and PRs largely aimed at improving legibility, fixing editorial errors, 
+   clearing up ambiguities, or adding examples to existing specifications. 
+   These may be progressed and merged without much process.
+
+2. **Contributions that _do_ meaningfully change the interpretation of the spec**  
+   These must progress through the stages of RFC, guided by a sponsor until 
+   they are ultimately accepted or rejected.
+
+## Contribution Stages
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: "Proposal Stages" or maybe we don't need a title here and just merge this with the section above

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,233 @@
...
+
+Sponsor organizations shall be transparently documented on the main repository.
+
+**Editors**
+
+Either working-group members or selected contributors responsible to repository maintenance: i.e have write/edit access
+to the main repository.
+
+
+## Classifying Contributions
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: This section title seems a bit off, Perhaps it should be renamed to "Proposals" (see also comments below about this)

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+RFCs are guided by a sponsor through a series of stages: **_explainer, proposal, draft, and accepted (or rejected)_**,
+each of which has suggested entrance criteria and next steps detailed below. RFCs typically advance one stage at a time
+but may advance multiple stages at a time. Stage advancements typically occur during Working Group meetings, but may
+also occur on GitHub.
+
+In general, it's preferable to start with a pull request so that we can best evaluate the RFC in detail. However,
+starting with an issue is also permitted if the full details are not worked out.
+
+All RFCs start as either an explainer or a proposal. They will be tagged with the appropriate issue/pr tag on GitHub as
+it progresses.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I'd remove "issue/pr"

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,233 @@
...
+
+### Stage 0: Explainer
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I'd rename these to "Explainer Stage" instead of having a numbering system

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,233 @@
...
+
+An RFC at the explainer stage captures a described problem or partially-considered solutions. An explainer does not need
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: "in the explainer stage"

> captures a described problem or partially-considered solutions

Wording seems a bit unclear here

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,233 @@
...
+to meet any entrance criteria. An explainer’s goal is to prove or disprove a problem and guide discussion towards either
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > entrance criteria

I'd mention that the entrance criteria are described later (or maybe add them in their own section before this)

> An explainer’s goal is to prove or disprove a problem

What does this mean?

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,233 @@
...
+rejection or a preferred solution. An explainer may be an issue or a pull request (though an illustrative pull request
+is preferrable).
+
+There are no entrance criteria for an explainer. It can simply be an issue, grievance or query.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I'd argue it should be well specified already, just potentially without a proper solution or not fully fleshed out

File: CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,233 @@
...
+
+As implied by the name explainer, the goal at this stage is to capture the issue, and from there, knock it down (reject)
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > implied by the name explainer
remove "explainer"

File: CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,233 @@
...
+by considering other possible related solutions, showing that the motivating problem can be solved with no change to the
+specification, or that it is not aligned with the guiding principles.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > knock it down (reject)
why not just say "reject it" instead of 

Also, acceptance is also an option but is not listed among the goals...

File: CONTRIBUTING.md [hunk 29]
```diff
@@ -0,0 +1,233 @@
...
+
+Once determined that the explainer is compelling, it should seek the entrance criteria for proposal.
+
+### Stage 1: Proposal
+
+An RFC at the proposal stage is a solution to a problem with enough fidelity to be discussed in detail. It must be
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: "fidelity"?

File: CONTRIBUTING.md [hunk 30]
```diff
@@ -0,0 +1,233 @@
...
+backed by a willing sponsor. A proposal's goal is to make a compelling case for acceptance by describing both the
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: Capitalize Sponsor

File: CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,233 @@
...
+* Identified sponsor
+* Identified target spec profile for the proposal
+    * A proposal may not add features to the core profile immediately, but rather pick an extension or module (or be a
+      module of its own). Core specifications may only change when an extension or module is migrated into core-spec via
+      a separate proposal.
+* Clear explanation of the problem and solution
+* Illustrative examples
+* .json Files for testing
+* Incomplete spec edits
+* Identification of potential concerns, challenges, and drawbacks
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > Identification of potential concerns, challenges, and drawbacks

Should this be done in the Explainer stage?

File: CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,233 @@
...
+
+A proposal is subject to the same discussion as an explainer: ensuring that it is well aligned with the guiding
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > explainer

I'd always capitalize stage names

File: CONTRIBUTING.md [hunk 33]
```diff
@@ -0,0 +1,233 @@
...
+principles, is a problem worth solving, and is the preferred solution to that problem. A sponsor is not expected to have
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > , is

I'd remove "is" or add "it" before "is"

File: CONTRIBUTING.md [hunk 34]
```diff
@@ -0,0 +1,233 @@
...
+confidence in every detail at this stage and should instead focus on identifying and resolving issues and edge-cases. To
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > edge-cases

"edge cases"

File: CONTRIBUTING.md [hunk 35]
```diff
@@ -0,0 +1,233 @@
...
+better understand the technical ramifications of the proposal, a sponsor is encouraged to implement it in a Lottie
+library.
+
+Most proposals are expected to evolve or change and may be rejected. Therefore, it is unwise to rely on a proposal in a
+production Lottie Library. Lottie libraries may implement proposals, though are encouraged to not enable the proposed
+feature without explicit opt-in.
+
+### Stage 2: Draft
+
+An RFC at the draft stage is a fully formed solution. There is working-group consensus that the problem identified
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > at the draft stage 

"in the Draft stage"

> working-group

"working group"

File: CONTRIBUTING.md [hunk 36]
```diff
@@ -0,0 +1,233 @@
...
+should be solved, and this particular solution is preferred. A draft's goal is to precisely and completely describe the
+solution and resolve any concerns through library implementations. **A draft must be a pull request.**
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: This is where the section chicken-or-egg section should go.

File: CONTRIBUTING.md [hunk 37]
```diff
@@ -0,0 +1,233 @@
...
+
+**Entrance criteria:**
+
+* Consensus the solution is preferred (typically via Working Group)
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > Consensus

How do we define that?

> (typically via Working Group)

Are there other ways?

File: CONTRIBUTING.md [hunk 38]
```diff
@@ -0,0 +1,233 @@
...
+* Resolution of identified concerns and challenges
+* Precisely described with spec edits
+* Relevant tests and test .json files
+* Compliant implementations:
+    * A library or tool of their choosing
+    * Machine readable Lottie-schema as hosted in this repository
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: I'd reword the last couple points as this:

```md
* Compliant implementations in an established library or tool of their choosing
* JSON-schema as hosted in the main repository
```

File: CONTRIBUTING.md [hunk 39]
```diff
@@ -0,0 +1,233 @@
...
+
+A proposal becomes a draft when the set of problems or drawbacks have been fully considered and accepted or resolved,
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > draft

Draft

> the set of problems or drawbacks

"potential drawbacks" maybe?

File: CONTRIBUTING.md [hunk 40]
```diff
@@ -0,0 +1,233 @@
...
+and the solution is deemed desirable. A draft's goal is to complete final spec edits that are ready to be merged and
+implement the draft in Lottie libraries along with tests to gain confidence that the spec text is sufficient.
+
+Drafts may continue to evolve and change, occasionally dramatically, and are not guaranteed to be accepted. Therefore,
+it is unwise to rely on a draft in a production Lottie Service. Lottie libraries should implement drafts to provide
+valuable feedback, though are encouraged not to enable the draft feature without explicit opt-in when possible.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: This is not really relevant for the person considering submitting a proposal. It should be in a compliance document.

File: CONTRIBUTING.md [hunk 41]
```diff
@@ -0,0 +1,233 @@
...
+
+A draft is accepted when the working group or editor (in the case of non invasive edits as outlined before) has been
+convinced via implementations and tests that it appropriately handles all edge cases; that the spec changes not only
+precisely describe the new syntax and semantics but include sufficient motivating prose and examples; and that the RFC
+includes edits to any other affected areas of the spec. Once accepted, its sponsor should encourage adoption of the RFC
+by opening issues or pull requests on other popular Lottie libraries.
+
+“Consensus” is defined as approval of two thirds of working-group. Proposals that call for or necessitate the movement
+of a feature previously regarded as an extension, into core-features, will require unanimous approval of the entire
+working-group.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: This should be defined before being referenced. I suggest having a "Definitions" section near the start of the document with this (and perhaps also "Sponsor" and maybe other terms used in the document

File: CONTRIBUTING.md [hunk 42]
```diff
@@ -0,0 +1,233 @@
...
+
+An accepted RFC is merged into the Lottie spec's main branch by an editor and will be included in the next released
+revision.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > the next released revision

"a future revision" I don't think we should guarantee merged proposals will always end up in the following standard version

File: CONTRIBUTING.md [hunk 43]
```diff
@@ -0,0 +1,233 @@
...
+
+### Stage X: Rejected
+
+An RFC may be rejected at any point. The reason for rejection should be clearly specified. If the proposal can be
+progressed after amendments, this avenue should be explored first.
```
Comment (mbasaglia) [inline CONTRIBUTING.md]: > If the proposal can be progressed after amendments, this avenue should be explored first.

It's not up to the working group to find ways of fixing rejected proposals. The burden falls on the person making the proposal

**Review by mbasaglia — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (kudanai) [reply CONTRIBUTING.md]: was wondering about that... I think it's in reference to the repository name "spec"

File: CONTRIBUTING.md (hunk 4 above)
Comment (kudanai) [reply CONTRIBUTING.md]: all the way at the bottom work?

File: CONTRIBUTING.md (hunk 12 above)
Comment (kudanai) [reply CONTRIBUTING.md]: changed the previous one to main repository

File: CONTRIBUTING.md (hunk 13 above)
Comment (kudanai) [reply CONTRIBUTING.md]: both right? the proposal itself and the individual submitting the proposal?

File: CONTRIBUTING.md (hunk 18 above)
Comment (kudanai) [reply CONTRIBUTING.md]: no we ca jump right into it without this bit

File: CONTRIBUTING.md (hunk 4 above)
Comment (mbasaglia) [reply CONTRIBUTING.md]: I mention a good spot in a different comment

File: CONTRIBUTING.md (hunk 23 above)
Comment (kudanai) [reply CONTRIBUTING.md]: or just Explainer? that takes care of that Stage X

File: CONTRIBUTING.md (hunk 13 above)
Comment (mbasaglia) [reply CONTRIBUTING.md]: dunno, the term "guiding" usually refers to people rather than things
